turbo/stages/headerdownload/header_algos.go

```diff
@@ -721,14 +723,19 @@ func (hi *HeaderInserter) FeedHeaderFunc(db kv.StatelessRwTx) func(header *types
 	return func(header *types.Header, hash common.Hash, blockHeight uint64) error {
 		return hi.FeedHeader(db, header, hash, blockHeight)
 	}
-
 }
 
 func (hi *HeaderInserter) FeedHeader(db kv.StatelessRwTx, header *types.Header, hash common.Hash, blockHeight uint64) error {
 	if hash == hi.prevHash {
 		// Skip duplicates
 		return nil
 	}
+
+	config, err := rawdb.ReadChainConfig(db, rawdb.ReadHeaderByNumber(db, 0).Hash())
```

Comment:
This will most probably be quite a lot of overhead - this function is called for each header. It is better to get the config when header inserter is created, or even better when stage is initialised